File: Examples/VisualBasic/GroupDocs.Assembly.Examples/CustomDocumentTableLoadHandler.vb

Imports GroupDocs.Assembly.Data

Public Class CustomDocumentTableLoadHandler
    Implements IDocumentTableLoadHandler
    Public Sub Handle(args As DocumentTableLoadArgs) Implements IDocumentTableLoadHandler.Handle
        Select Case args.TableIndex
            Case 0
                ' Do nothing. The table is to be loaded with default options.
                Exit Select
            Case 1
                ' Discard loading of the table completely.
                args.IsLoaded = False
                Exit Select
            Case 2
                ' Load the table with custom options.
                args.Options = New DocumentTableOptions()
                args.Options.FirstRowContainsColumnNames = True
                Exit Select
            Case Else
                Throw New InvalidOperationException()
        End Select
    End Sub

End Class

File: Examples/CSharp/GroupDocs.AssemblyExamples/CustomDocumentTableLoadHandler.cs
using GroupDocs.Assembly.Data;
using System;

namespace GroupDocs.AssemblyExamples
{
    public class CustomDocumentTableLoadHandler : IDocumentTableLoadHandler
    {
        public void Handle(DocumentTableLoadArgs args)
        {
            switch (args.TableIndex)
            {
                case 0:
                    // Do nothing. The table is to be loaded with default options.
                    break;
                case 1:
                    // Discard loading of the table completely.
                    args.IsLoaded = false;
                    break;
                case 2:
                    // Load the table with custom options.
                    args.Options = new DocumentTableOptions();
                    args.Options.FirstRowContainsColumnNames = true;
                    break;
                default:
                    throw new InvalidOperationException();
            }
        }
    }
}
